terraform/tfschema/validators.go

```diff
@@ -163,3 +164,67 @@ OUTER:
 	}
 
 }
+
+// OneOfValidator validates that at least one of the attributes is set (known and not null)
+type OneOfValidator struct {
+	Keys []string
+}
+
+// UseOneOfValidator creates OneOfValidator
+func UseOneOfValidator(keys ...string) tfsdk.AttributeValidator {
+	return OneOfValidator{
+		Keys: keys,
+	}
+}
+
+// Description returns validator description
+func (v OneOfValidator) Description(_ context.Context) string {
+	return fmt.Sprintf("OneOf '%s' attributes must be present", strings.Join(v.Keys, ", "))
+}
+
+// MarkdownDescription returns validator markdown description
+func (v OneOfValidator) MarkdownDescription(_ context.Context) string {
+	return fmt.Sprintf("OneOf `%s` attributes must be present", strings.Join(v.Keys, ", "))
+}
+
+// Validate performs the validation.
+func (v OneOfValidator) Validate(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+	if req.AttributeConfig == nil {
+		return
+	}
+
+	value, ok := req.AttributeConfig.(types.Object)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"OneOf Object validation error",
+			fmt.Sprintf("Attribute %v can not be converted to Object", req.AttributePath.String()),
+		)
+		return
+	}
+
+	if value.Null || value.Unknown {
+		return
+	}
+
+	for _, key := range v.Keys {
```

Comment:
I may be missing something but I don't see where in this function we set the error if both fields are set? Shouldn't they be mutually exclusive?